### PR TITLE
Allow specifying memory, cpu and vcpu as kwargs for OpenNebula

### PR DESCRIPTION
**Pull request by fracklen**
### What does this PR do?

Allow specifying memory, cpu and vcpu as kwargs for OpenNebula
### What issues does this PR fix or reference?

None - I've used it for some time and want to contribute back.
### New Behavior

Allows cloud profile for OpenNebula to define
- memory
- cpu
- vcpu

To alter the defaults specified in the template.
### Tests written?

No - The test for OpenNebula hasn't got proper mocking for the API calls, and as such, tests have not been written for the create method.

Why:

When instantiating a VM in OpenNebula, this enables a single template to
be used in multiple scenarios, simply by overwriting the dimensions
(memory, cpu and vcpu) of the template.

It also clears up the potential confusion, that the 5th argument to
template.instantiate should be a region specification. Based on the docs
http://docs.opennebula.org/5.0/integration/system_interfaces/api.html#one-template-instantiate
it is a partial template which allows overwriting parts of the real template.

### CODE REVIEW

File: salt/cloud/clouds/opennebula.py [hunk 1]
```diff
@@ -901,17 +917,24 @@ def create(vm_):
         {'kwargs': kwargs},
     )
 
-    region = ''
+    template = []
     if kwargs['region_id'] is not None:
-        region = 'SCHED_REQUIREMENTS="ID={0}"'.format(kwargs['region_id'])
+        template.append('SCHED_REQUIREMENTS="ID={0}"'.format(kwargs['region_id']))
+    if vm_['memory'] is not None:
```
Comment (rallytime) [inline salt/cloud/clouds/opennebula.py]: Can you change this to a `vm_.get('memory')` call? That way we can avoid a KeyError if the vm_ passed in doesn't have this option and will default to `None`. The same comment goes for all of the options you've added below.

**Comment by rallytime** [comment]
@fracklen Thank you for this contribution. I had one comment that needs to be addressed, then we can get this in.

**Comment by fracklen** [comment]
@rallytime fixed - great!

**Comment by rallytime** [comment]
Much cleaner - thanks!

**Pull request merged**